### DISCUSSION TITLE
Is safe use ReadOnlyMemory across lists?

### DISCUSSION THREAD

**Discussion by nathan130200** [dotnet/runtime#80307] [Q&A] [answered]
I'm working on an send queue to an TCP socket, i will store buffer to send and task completition source, to notify if send is completed or got some error.

I was using raw byte array in queue entry type. But i'm finding solutions to reduce memory usage.

```cs
class QueueEntry {
  public byte[] Data { get; init; }
  public TaskCompletitionSource Completition { get; init; }
}
```

If i set this `Data` to an existing buffer will peform an copy or will still pass array by reference?
Instead i can use `ReadOnlyMemory<byte>` to replace `byte[]`? Since its value type will pass by reference too without data loss?

**Comment by FaustVX** [comment, marked as answer]
The (RO)Memory, (like the (RO)Span) just contains the reference to the array, not a copy
